packages/shared/src/authz/domain/errors.ts

```diff
@@ -0,0 +1,36 @@
+type AuthError<ErrorCode extends string> = {
```

Comment:
I removed the `M extends string` bit in the generic for the message because the types were coming through _wide_ like:

`AuthError<"ProjectNotFound", "Project not found">`